client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,202 @@
+import React, { useMemo, useState } from 'react';
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Form,
+  Input,
+  FormGroup,
+  InputGroupAddon
+} from 'reactstrap';
+import { LoadingSpinner, FormField, Icon, InlineMessage } from '_common';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Formik } from 'formik';
+import PropTypes from 'prop-types';
+import * as yup from 'yup';
+import './DataFilesCompressModal.module.scss';
+
+const DataFilesCompressForm = ({
+  singleFile,
+  formRef,
+  isDisabled,
+  validator
+}) => {
+  const initialValues = {
+    filenameDisplay: `${singleFile || ''}`,
+    filetype: '.zip'
+  };
+  const validationSchema = yup.object().shape({
+    filenameDisplay: yup
+      .string()
+      .trim('The filename cannot include leading and trailing spaces')
+      .strict(true)
+      .required('The filename is required')
+  });
+
+  return (
+    <Formik
+      innerRef={formRef}
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+    >
+      {({ values, setFieldValue, isValid }) => {
+        validator(isValid);
+        const handleSelectChange = e => {
+          setFieldValue('filetype', e.target.value);
+        };
+        return (
+          <Form>
+            <FormGroup>
+              <FormField
+                label="Filename"
```

Comment:
**Text Change** (from Design)

> Pending confirmation. Do **not** "Commit suggestion" yet.

There has been this text change defined in [a message in private channel `cep-design-internal`](https://tacc-team.slack.com/archives/G01K9ET7JM7/p1610671076006400).

```suggestion
                label="[pending confirmation of change to this text]"
```

Reply:
**Update**

Text change confirmed: "Filename" → "Compressed File Name"